glean-core/csharp/GleanTests/Metrics/PingTypeTest.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.IO;
+using Xunit;
+using static Mozilla.Glean.Glean;
+
+namespace Mozilla.Glean.Tests.Metrics
+{
+    public class PingTypeTest
+    {
+        public PingTypeTest()
+        {
+            // Get a random test directory just for this single test.
+            string tempDataDir = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+
+            // In xUnit, the constructor will be called before each test. This
+            // feels like a natural place to initialize / reset Glean.
+            GleanInstance.Reset(
+                applicationId: "org.mozilla.csharp.tests",
+                applicationVersion: "1.0-test",
+                uploadEnabled: true,
+                configuration: new Configuration(),
+                dataDir: tempDataDir
+                );
+        }
+
+        [Fact]
+        public void SendCustomPings()
```

Comment:
You can port over all the tests defined [here](https://searchfox.org/glean/source/glean-core/android/src/test/java/mozilla/telemetry/glean/private/PingTypeTest.kt). Unfortunately, almost all of them require a mock web server and an uploader: I'm fine with you just implementing [this one](https://searchfox.org/glean/rev/b7c91243657df02f3ef2d579d02b27661c23e54e/glean-core/android/src/test/java/mozilla/telemetry/glean/private/PingTypeTest.kt#229) and leave a TODO for the others.

Reply:
I still have problems on it. Let's discuss it at #965